blaze/server/tests/test_server.py

```diff
@@ -660,24 +588,59 @@ def test_cant_add_data_to_server(iris_server, serial):
     # try adding more data to server
     iris_path = example('iris.csv')
     blob = serial.dumps({'iris': iris_path})
-    response1 = iris_server.post(
-        '/add',
-        headers=mimetype(serial),
-        data=blob,
-    )
-    assert response1.status_code == 422
+    response1 = iris_server.post('/add',
+                                 headers=mimetype(serial),
+                                 data=blob)
+    assert response1.status_code == RC.UNPROCESSABLE_ENTITY
+
+
+@pytest.mark.parametrize('serial', all_formats)
+def test_add_data_twice_error(empty_server, serial):
+    with temp_server():
+        # add iris
+        iris_path = example('iris.csv')
+        payload = serial.dumps({'iris': iris_path})
+        empty_server.post('/add',
+                          headers=mimetype(serial),
+                          data=payload)
+        # Try to add to existing 'iris'
+        resp = empty_server.post('/add',
+                                 headers=mimetype(serial),
+                                 data=payload)
```

Comment:
The `CONFLICT` is triggered on the name of the dataset, not the contents; I'm fine if we keep the payload the same.

I'll add the test that the server still serves the original.